### ISSUE TITLE
some question during training and achieve your paper  result

### ISSUE THREAD

**Issue by JcWang20** [ljwztc/CLIP-Driven-Universal-Model#73]
Why does the postlabel file you open only have 31 channels after reading it, but doesn’t it have a total of 32 categories? In addition, during the training process, I found that the dice loss could not be reduced. In the end, even after training for 500 epochs, I could not achieve the results in the paper. Especially on the sub-task of msd, it was far lower than your pretrain model. On the other hand, I found that after datalodaer read The data shape is b, c, h, w, d, but I understand that your code is used according to the shape b, c, d, h, w. So is there any reason for this?
Looking forward to your reply as soon as possible

**Comment by ljwztc** [comment]
1. The issue with 31 and 32 channels is due to the different data processing setting (whether there is label 32 Kidney Cyst). If you load one dataset on one GPU and only increase the --num_samples parameter, the code should run smoothly. However, if you want to load more data on one GPU, you may need to process the post-label data yourself with same setting.

2. When we trained the model, the Dice loss value is roughly around 0.8 and does not decrease significantly lower than that. 

3. I think package version is a very important but not only factors to cause the result difference. 

4. If you keep the code in dataset/dataloader.py at lines 229-236 original, the loaded shapes for the image and H5 label data should be the same. This assumes that you have followed the instructions in the README strictly. Additionally, using different versions of the MONAI library may also cause this issue.

**Comment by JcWang20** [comment]
> 1. The issue with 31 and 32 channels is due to the different data processing setting (whether there is label 32 Kidney Cyst). If you load one dataset on one GPU and only increase the --num_samples parameter, the code should run smoothly. However, if you want to load more data on one GPU, you may need to process the post-label data yourself with same setting.
I hava downloaded and processed the 12 data sets mentioned in this repo according to your settings, but I was unable to train and get a result mentioned in the paper. Especially on btcv and msd, the difference between the results and the dice score mentioned in the paper exceeds 10 points, no matter whether I use the post_label you released or the postlabel I got according to the label transfer file you provided, I can't get normal results.
It is worth mentioning that I used all the CT data mentioned in the paot and pato inner 2 txts for training, and added the training stage from task01 to task12, 8 node a100 training for 500epoch, but I could not get the dice as high as mentioned in the paper. The results, especially on msd and btcv, are far lower than directly using the pretrained model you provided for testing.
At the same time, I sent you an email and did not get a reply from you. If possible, I hope to communicate with you in more detail about this codebase, because I think your work is very meaningful, and the code processing of the entire codebase is also relatively complete. At the same time, I It also took a lot of effort and time to collect data and try to train your code, so I want to solve this problem and continue to follow your work

**Comment by ljwztc** [comment]
Thank you very much for your appreciation of our work. We are delighted to receive positive feedback. Please note that we receive numerous inquiries regarding the code, and due to our workload, we respond to these queries in intervals with certain time. We kindly request your patience, and rest assured, we will address your inquiry as soon as possible. May I inquire if you are using the same package version as us? We have encountered the problem that different versions of monai could effect in the our subsequent work.
